### PR TITLE
Adds unit tests

### PR DESCRIPTION
**Pull request by fealho**
Adds unit tests to the synthesizers and DataTransformer, as well as some other minor improvements in code readability.

### CODE REVIEW

File: tests/unit/synthesizer/test_ctgan.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        for parameter in residual.parameters():
+            assert parameter.grad is not None
+
+
+class TestGenerator(TestCase):
+
+    def test___init__(self):
+        """Test `__init__` for a generic case.
+
+        Make sure 'self.seq' has same length as `generator_dim` + 1.
```
Comment (pvk-developer) [inline tests/unit/synthesizer/test_ctgan.py:115]: 'self.seq' -> `self.seq`

File: tests/unit/test_data_transformer.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+            [0, 1],
+            [1, 0],
+        ])]
+
+        result = transformer.transform(data)
+        transformer._transform_continuous.assert_called_once()
+        transformer._transform_discrete.assert_called_once()
+        assert result.shape == (3, 6)
+        assert (result[:, 0] == np.array([0.1, 0.3, 0.5])).all()
+        assert (result[:, 1:4] == np.array([
```
Comment (pvk-developer) [inline tests/unit/test_data_transformer.py:245]: I suggest we use variable for the expected result on one forth, similar to this:
```python
        one_forth_expected = np.array([
            [1, 0, 0],
            [1, 0, 0],
            [1, 0, 0],
        ])
        assert (result[:, 1:4] == one_forth_expected).all()
```
It's easier to read and we don't have that multi-line.

File: tests/unit/test_data_transformer.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+        ]
+
+        transformer._transform_continuous = Mock()
+        selected_normalized_value = np.array([[0.1], [0.3], [0.5]])
+        selected_component_onehot = np.array([
+            [1, 0, 0],
+            [1, 0, 0],
+            [1, 0, 0],
+        ])
+        transformer._transform_continuous.return_value = (selected_normalized_value,
```
Comment (pvk-developer) [inline tests/unit/test_data_transformer.py:230]: Let's use one extra variable and lose that extra line that we create below.
```python
        continuous_return = (selected_normalized_value, selected_component_onehot)
        transformer._transform_continuous.return_value = continuous_return
```

File: tests/unit/test_data_transformer.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+        transformer._transform_continuous.assert_called_once()
+        transformer._transform_discrete.assert_called_once()
+        assert result.shape == (3, 6)
+        assert (result[:, 0] == np.array([0.1, 0.3, 0.5])).all()
+        assert (result[:, 1:4] == np.array([
+            [1, 0, 0],
+            [1, 0, 0],
+            [1, 0, 0],
+        ])).all()
+        assert (result[:, 4:6] == np.array([
```
Comment (pvk-developer) [inline tests/unit/test_data_transformer.py:250]: Same as above.

File: tests/unit/test_data_transformer.py [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+                output_dimensions=1 + 3
+            ),
+            ColumnTransformInfo(
+                column_name='y', column_type='discrete', transform=ohe,
+                transform_aux=None,
+                output_info=[SpanInfo(2, 'softmax')],
+                output_dimensions=2
+            )
+        ]
+        result = transformer.convert_column_name_value_to_id('y', 'yes')
```
Comment (pvk-developer) [inline tests/unit/test_data_transformer.py:349]: add a white space to keep the same style where we split the asserts and run results. (as below)

**Review by pvk-developer — APPROVED** [review]

**Review by csala — APPROVED** [review]

**Pull request merged**